src/libm/sleefsimddp.c

```diff
@@ -2783,4 +2783,7 @@ EXPORT CONST vdouble __sinh_finite     (vdouble)          __attribute__((weak, a
 EXPORT CONST vdouble __sqrt_u05_finite (vdouble)          __attribute__((weak, alias(str_xsqrt_u05 )));
 EXPORT CONST vdouble __tgamma_u1_finite(vdouble)          __attribute__((weak, alias(str_xtgamma_u1)));
 
+#ifdef HEADER_MASKED
+#include HEADER_MASKED
```

Comment:
@shibatch , thank you for working on this. 
I'd prefer having the explicit functions here, with a "masked" name being generated by the `mkrename_gnuabi` tool.

For example:
```
#ifdef ENABLE_MASKED_INTERFACE
vdouble xcos_masked(vdouble, vopmask) {
  // function body
  return xcos(vdouble); // temporary implementation
}
// ...
#endif

```
The reason for doing so is that for now we are just using the original unmasked functions to execute the masked version, but in the future we will come up with fully-masked implementations, using masked intrinsics. It will be easier to extend this work than to deal with the header file.

Also, I think we should enable the masked symbols only for those vector extension that support masking natively, i.e. AVX512F.

Reply:
If we implement in that way, we need to maintain many hand-written functions, which would be troublesome. I think automatically generating the header is easier to maintain and less bug-prone.

Also, it is unclear when we will start implementation of the body of masked functions. It is not too late to convert the header to fully-implemented functions at that time.

Reply:
I disagree with your "easier to maintain and less bug-prone".

The thing is that the masked functions will probably have to be replaced with a proper implementation as soon as someone start to use them, because they will be not as good as they could be when implemented with native predication. I also expect such replacement to happen gradually, function by function, not with a single commit: in that case, it will be easier to modify the functions in place instead of having to deal with the auto generative process.

Reply:
Update: I am happy with the idea of the auto-generated interface. We will work on a new version according to #142 .